libs/sdk-core/src/models.rs

```diff
@@ -132,63 +137,137 @@ pub struct ReverseSwapPairInfo {
 }
 
 /// Details of past or ongoing reverse swaps, as stored in the Breez local DB
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub struct ReverseSwapInfo {
-    // static immutable data
-    pub claim_address: String,
+    /// The reverse swap ID, as reported by the Boltz API in case of a successful creation
+    pub id: String,
+
+    /// Local timestamp when the reverse swap is created
+    pub created_at: i64,
+
+    /// Locally generated preimage, revealed in the last step of the reverse swap
+    pub local_preimage: Vec<u8>,
+
+    /// Locally generated private key, used to sign the claim tx
+    pub local_private_key: Vec<u8>,
+
+    /// On-chain destination address, to which the reverse swap will finally send funds to
+    pub destination_address: String,
+
+    /// Reverse swap status, as reported by the Boltz API. Automatically updated on every new block.
+    pub boltz_api_status: BoltzApiReverseSwapStatus,
+
+    // Only available in the first response from Boltz API:
+    pub hodl_bolt11: String,
+    pub redeem_script: String,
+
+    pub cache: ReverseSwapInfoCached,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ReverseSwapInfoCached {
     pub lockup_address: String,
+    pub onchain_amount_sat: u64,
+}
 
-    // dynamic data
-    pub status: ReverseSwapStatus,
+impl ReverseSwapInfo {
+    /// Dynamically determine the status, based on the cached Boltz API status.
+    ///
+    /// In case of [BoltzApiReverseSwapStatus::SwapCreated], the payment status is looked up on the LN node,
+    /// to distinguish between a reverse swap that is still ongoing and one where the HODL payment failed.
+    pub async fn status(&self, node_api: Arc<dyn NodeAPI>) -> Result<ReverseSwapStatus> {
+        let status = match &self.boltz_api_status {
+            BoltzApiReverseSwapStatus::SwapCreated => {
+                debug!("Looking up payment via NodeAPI");
+                let invoice = parse_invoice(&self.hodl_bolt11)?;
+                let payment_hash = hex::decode(invoice.payment_hash)?;
+                match node_api.lookup_payment(payment_hash).await? {
+                    None => ReverseSwapStatus::FailedToPayInvoice, // Was aborted before pay() could throw error
+                    Some(p) => match p.pending {
+                        true => ReverseSwapStatus::Initial, // Payment is in progress
+                        false => ReverseSwapStatus::FailedToPayInvoice, // The node reports it as failed
```

Comment:
Why failed? Not pending might also mean payment succeed.

Reply:
Normally yes, but for the HODL invoice not the case because the preimage is not shared.

This invoice is settled (not pending anymore) when we publish the ClaimTx, which reveals the preimage to Boltz, which they then use to settle the invoice.